### PR TITLE
Fix hold fire interactions

### PR DESCRIPTION
**Pull request by warp-core**
**Bug fix**

This PR addresses the bug/feature described in issue #10578

## Summary
Fixes the bug described in the linked issue by handling the case of the only order being to hold fire in `AI::FollowOrders`.

## Testing Done
Using the "Farpoint" save from the linked issue, take off, order escorts to hold fire, order them to gather around me, and then order them to stop gathering. With this PR, they act correctly and don't leave the system.

## Save File
This save file can be used to test these changes:
See the linked issue.

### CODE REVIEW

File: source/Orders.h [hunk 1]
```diff
@@ -78,6 +78,7 @@ class Orders {
 	bool HasHarvest() const;
 
 	bool IsEmpty() const;
+	size_t count() const;
```
Comment (tibetiroka) [inline source/Orders.h:81]: Should this be capitalized? We already use `IsEmpty` instead of `empty`.
```suggestion
	size_t Count() const;
```

Comment (warp-core) [reply source/Orders.h:81]: If we consider this class to be an extension of `std::bitset`, then I think it should be lower-case.
`std::bitset` has no `empty` method.

Comment (tibetiroka) [reply source/Orders.h:81]: It does have `any`, though.

**Review by RisingLeaf — APPROVED** [review]

**Review by tibetiroka — CHANGES_REQUESTED** [review]
Doesn't work. Order your escorts to move to a specific location, then tell them to hold fire; they move to the center of the system and stay there until you give them another move order.

**Comment by warp-core** [comment]
> Doesn't work. Order your escorts to move to a specific location, then tell them to hold fire; they move to the center of the system and stay there until you give them another move order.

This is a (separate) existing problem not impacted by the changes in this PR, and not described in the bug report. Consider opening a new issue.

**Comment by warp-core** [comment]
I don't like this PR. I think the changes are not very robust. I have a feeling they actually introduce new issues, but it's midnight and I don't feel like doing more testing right now. I'm seriously considering reverting de787b4 and 09bb24b..

**Comment by tibetiroka** [comment]
Closing in favor of #10652.

**Pull request closed without merging**